File: Server/src/models/responses/apiResponse.js
class ApiResponse {
    constructor({ isError = false, message = "", data = null, statusCode = 200 }) {
        this.isError = isError;
        this.message = message;
        this.data = data;
        this.statusCode = statusCode;
    }

    static success(message, data = null, statusCode = 200) {
        return new ApiResponse({ isError: false, message, data, statusCode });
    }

    static error(message, statusCode = 500) {
        return new ApiResponse({ isError: true, message, data: null, statusCode });
    }
}

module.exports = ApiResponse;

File: lib/core/constants/api_response.dart
class ApiResponse<T> {
  bool? success;
  String? message;
  T? data;

  ApiResponse({this.success, this.message, this.data});

  factory ApiResponse.fromJson(Map<String, dynamic> json, Function? fromJson) {
    return ApiResponse<T>(
      success: json["success"],
      message: json["message"],
      data: json["data"] != null && fromJson != null ? fromJson(json["data"]) : null,
    );
  }

  Map<String, dynamic> toJson(Function? toJson) {
    return {
      "success": success,
      "message": message,
      "data": data != null && toJson != null ? toJson(data!) : null,
    };
  }
}
